src/app/hooks/usePermissions/types.ts

```diff
@@ -0,0 +1,54 @@
+// Core types for ABAC (Attribute-Based Access Control) system
+
+export type Action = "view" | "create" | "update" | "delete" | "approve" | "assign" | "cancel"
```

Comment:
I think action, role, and resource should be string types instead of explicitly defined strings, because as we build out the application and add new resources, roles, and actions, we're going to forget to update these types.

Instead if its just string types, the only place that needs to be updated is the permissions, otherwise if the role or resource doesnt exist it can return false and console log so the missing permission or typo can be fixed during development.